### PR TITLE
feat(ingestion/powerbi): Ingest datasets not used in PowerBI visualization(tiles/pages)

### PR DESCRIPTION
**Pull request by mohdsiddique**

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py [hunk 1]
```diff
@@ -382,11 +384,14 @@ def _fill_metadata_from_scan_result(
...
                 dashboard_endorsements={},
                 scan_result={},
+                independent_datasets=[],
             )
             cur_workspace.scan_result = workspace_metadata
             cur_workspace.datasets = self._get_workspace_datasets(
                 cur_workspace.scan_result
             )
+            for value in cur_workspace.datasets.values():
+                logger.info(f"Mohd = {value.name}")
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py:394]: remove this

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py [hunk 2]
```diff
@@ -1179,11 +1203,8 @@ def get_workspace_workunit(
             ):
                 yield work_unit
 
-        for dataset in self.mapper.processed_datasets:
-            if self.source_config.extract_datasets_to_containers:
-                dataset_workunits = self.mapper.generate_container_for_dataset(dataset)
-                for workunit in dataset_workunits:
-                    yield workunit
+        yield from self.extract_datasets_as_containers()
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py:1206]: i'd like to keep the `self.source_config.extract_datasets_to_containers` check here

File: metadata-ingestion/tests/integration/powerbi/golden_test_independent_datasets.json
Comment (hsheth2) [inline metadata-ingestion/tests/integration/powerbi/golden_test_independent_datasets.json:1]: these tests are getting pretty unwieldy - is there a way we can reduce the duplication across all the golden files?

**Review by hsheth2 — CHANGES_REQUESTED** [review]
mostly looking good, just a minor code cleanup thing

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py (hunk 1 above)
Comment (mohdsiddique) [reply metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py:394]: done

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py (hunk 2 above)
Comment (mohdsiddique) [reply metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py:1206]: done

File: metadata-ingestion/tests/integration/powerbi/golden_test_independent_datasets.json
Comment (mohdsiddique) [reply metadata-ingestion/tests/integration/powerbi/golden_test_independent_datasets.json:1]: I remove the unwanted mock data. Now this file size got reduced to 2k

**Review by hsheth2 — APPROVED** [review]

**Comment by hsheth2** [comment]
Merging because these changes aren't related to the kafka build

**Pull request merged**